lib/ecco/event_listener.rb

```diff
@@ -3,15 +3,33 @@ class EventListener
     include com.github.shyiko.mysql.binlog.BinaryLogClient::EventListener
     java_import com.github.shyiko.mysql.binlog.event.EventType
 
+    WRITE_EVENTS  = { EventType::WRITE_ROWS => "WRITE_ROWS", EventType::EXT_WRITE_ROWS => "WRITE_ROWS" }
+    UPDATE_EVENTS = { EventType::UPDATE_ROWS => "UPDATE_ROWS", EventType::EXT_UPDATE_ROWS => "UPDATE_ROWS" }
+    DELETE_EVENTS = { EventType::DELETE_ROWS => "DELETE_ROWS" , EventType::EXT_DELETE_ROWS => "DELETE_ROWS" }
+    QUERY_EVENTS  = { EventType::QUERY => "QUERY"}
+    ROTATE_EVENTS = { EventType::ROTATE => "ROTATE"}
+
+    TABLE_EVENT = EventType::TABLE_MAP
+
     attr_writer :callback
 
-    def initialize(client)
+    def initialize(client, events_of_interest)
       @client = client
+      @event_list = events_of_interest.reduce(&:merge)
+      @accepted_events = @event_list.keys
       @callback = Proc.new {}
     end
```

Comment:
One final suggestion, not sure if the new variable name is that much better than `event_list` though, but I like the grouping I did below at least :)

```diff
diff --git a/lib/ecco/event_listener.rb b/lib/ecco/event_listener.rb
index 235aba7..fc01849 100644
--- a/lib/ecco/event_listener.rb
+++ b/lib/ecco/event_listener.rb
@@ -15,8 +15,10 @@ module Ecco
 
     def initialize(client, events_of_interest)
       @client = client
-      @event_list = events_of_interest.reduce(&:merge)
-      @accepted_events = @event_list.keys
+
+      @events_of_interest = events_of_interest.reduce(&:merge)
+      @accepted_events    = @events_of_interest.keys
+
       @callback = Proc.new {}
     end
 
@@ -29,7 +31,7 @@ module Ecco
     end
 
     def event_type_to_string(type)
-      @event_list.fetch(type)
+      @events_of_interest.fetch(type)
     end
   end
 end
```

Reply:
Nice! c7c0a5b